core/src/proof.rs

```diff
@@ -53,6 +78,18 @@ impl std::fmt::Display for DataIntegrityProof {
     }
 }
 
+impl ProofOptionDocument {
```

Comment:
Any reason not to just implement `From<ProofOptionDocument>` for `DataIntegrityProof`?

Reply:
@gshuflin This requires an input, its not a pure conversion so we can't do `From<ProofOptionDocument>` since that requires no parameters to be passed?

Reply:
Unless you know of some workaround that I'm not aware of